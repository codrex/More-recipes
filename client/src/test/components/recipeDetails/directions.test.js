/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import toJson from 'enzyme-to-json';
import Directions from '../../../components/pages/RecipeDetails/Directions';

const props = {
  directions: ['directions 1', 'direction 2', 'direction 3'],
};


describe('Directions display component ', () => {
  test('expected to render a list of direction', () => {
    const component = mount(<Directions {...props} />);
    const tree = toJson(component);
    expect(component.find('Accordion').length).toBe(3);
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });

  test('should open recipe accordion when show button is clicked',
    () => {
      const component = mount(<Directions directions={['directions 1']} />);
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
      component.find('a#toggle').simulate('click');
      expect(component.find('Direction').instance().state.open).toBe(true);
      expect(tree).toMatchSnapshot();
      expect(tree).toBeInstanceOf(Object);
    });
});
