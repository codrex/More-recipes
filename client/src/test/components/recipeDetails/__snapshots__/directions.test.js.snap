// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directions display component  expected to match empty snapshot 1`] = `
<div
  className="col-12 directions-list-wrapper"
>
  <h5
    className="display-4"
  >
    Directions 
  </h5>
  <Direction
    content="directions 1"
    index={0}
    key="directions 11514970600132"
  />
  <Direction
    content="direction 2"
    index={1}
    key="direction 21514970600132"
  />
  <Direction
    content="direction 3"
    index={2}
    key="direction 31514970600132"
  />
</div>
`;
